src/execute_command.rs

```diff
@@ -62,84 +70,130 @@ impl Stdin {
     }
 }
 
-impl Output {
+impl From<File> for Stdin {
+    fn from(file: File) -> Self {
+        Stdin::File(file)
+    }
+}
+
+#[cfg(unix)]
+impl AsRawFd for Stdin {
+    fn as_raw_fd(&self) -> RawFd {
+        use libc;
+
+        match self {
+            Stdin::Inherit => libc::STDIN_FILENO,
+            Stdin::File(f) => f.as_raw_fd(),
+            Stdin::FileDescriptor(fd) => *fd,
+            Stdin::Child(child) => child.as_raw_fd(),
+        }
+    }
+}
+
+impl Stdout {
     /// simple commands prefer file redirects to piping, following bash's behavior
     #[cfg(unix)]
-    fn new(redirect: &ir::Stdio, pipe: Option<Output>) -> Result<Self> {
-        use std::os::unix::io::FromRawFd;
-
+    fn new(redirect: &ir::Stdio, pipe: Option<Stdout>) -> Result<Self> {
```

Comment:
actually it looks like it would be simpler to just rewrite the `new` functions on `Output` to have `from_stdout` and `from_stderr` instead of introducing a whole new type. At this point, the new type isn't necessary.